core/src/test/java/hudson/FilePathTest.java

```diff
@@ -325,6 +328,33 @@ private FilePath createFilePath(final File base, final String... path) throws IO
         FileUtils.touch(building);
         return new FilePath(building);
     }
+    
+    
+    /**
+     * Checks that big files (>8GB) can be archived and then unpacked.
+     * This test is disabled by default due the impact on RAM.
+     * The actual file size limit is 8589934591 bytes.
+     * @throws Exception test failure
+     */
+    @Issue("JENKINS-10629")
+    @Ignore
+    @Test public void archiveBigFile() throws Exception {
+        assumeTrue(!Functions.isWindows());
+        long largeFileSize = 10000000000l;
+ 
+        File tempFile = File.createTempFile("JENKINS-10629", ".log");
+        RandomAccessFile file = new RandomAccessFile(tempFile, "rw");
+        File tarFile = new File(tempFile.getParent(), "JENKINS-10629.tar");
+        tarFile.deleteOnExit();
+        tempFile.deleteOnExit();
+ 
+        file.setLength(largeFileSize);
+        assumeTrue(largeFileSize == file.length());
+ 
+        int tar = new FilePath(tempFile.getParentFile()).tar(new FileOutputStream(tarFile), tempFile.getName());
+ 
+        assertEquals("One tar file should have been compressed", 1, tar);
```

Comment:
nit: its not the tar file that is being compressed (and nothing is being compressed).  Its one file file being archived (or tarred)